### PR TITLE
MM-15220 Document Interactive Dialog submission error field

### PR DESCRIPTION
**Pull request by pbitty**
This adds the documentation for the feature being implemented in https://github.com/mattermost/mattermost-server/issues/12043.

#### Ticket links

https://mattermost.atlassian.net/browse/MM-15220

### CODE REVIEW

File: source/developer/interactive-dialogs.rst [hunk 1]
```diff
@@ -274,6 +274,12 @@ Moreover, Mattermost also allows the integration itself to perform input validat
 
   {"errors": {"num_between_0_and_10": "Enter a number between 0 and 10."}}
 
+The integration may also return a generic error message to the user that is not attached to a specific field. This can be done by responding to the dialog submission request with a JSON body containing an `error` field.  The `error` field should contain a string with the error message to display to the user.  For example, if a server-side error occurs, you can return a message explaining it:
+
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: Would you please add a note about the required server version for this? Something like:
```suggestion

	This was added in Mattermost 5.16

```

File: source/developer/interactive-dialogs.rst [hunk 2]
```diff
@@ -274,6 +274,12 @@ Moreover, Mattermost also allows the integration itself to perform input validat
...
+.. code-block:: json
+
+  {"error": "Some of your inputs are invalid.  Please check your inputs and try again."}
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: I think that not the greatest of examples. Maybe something like
```suggestion
  {"error": "Failed to fetch additional data. Please try again."}
```

**Comment by hanzei** [comment]
Thanks for taking care of the documentation @pbitty :+1:

File: source/developer/interactive-dialogs.rst (hunk 1 above)
Comment (pbitty) [reply source/developer/interactive-dialogs.rst]: Done.  I added it just after the JSON example.

**Comment by pbitty** [comment]
@hanzei No prob!  Feedback addressed.

**Comment by hanzei** [comment]
@amyblais Would you be fine to merge this after the reviews, given that it's clearly documented that this will be in 5.16?

**Comment by amyblais** [comment]
@hanzei I think it's fine. Is there a specific reason why we want to get this merged before v5.16 is released?

File: source/developer/interactive-dialogs.rst [hunk 3]
```diff
@@ -274,6 +274,14 @@ Moreover, Mattermost also allows the integration itself to perform input validat
 
   {"errors": {"num_between_0_and_10": "Enter a number between 0 and 10."}}
 
+The integration may also return a generic error message to the user that is not attached to a specific field. This can be done by responding to the dialog submission request with a JSON body containing an `error` field. The `error` field should contain a string with the error message to display to the user. For example, if a server-side error occurs, you can return a message explaining it:
+
+.. code-block:: json
+
+  {"error": "Failed to fetch additional data. Please try again."}
+
+Support for generic error messages was added in Mattermost 5.16.
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: @pbitty This feature won't make it into 5.16. Would you please change this to 5.18?
```suggestion
Support for generic error messages was added in Mattermost 5.18.
```

**Comment by hanzei** [comment]
> I think it's fine. Is there a specific reason why we want to get this merged before v5.16 is released

@amyblais This allows integration developers to already learn about this feature and use it.

**Comment by amyblais** [comment]
Thank you, it should be fine,

**Comment by hanzei** [comment]
@aaronrothschild Would you please give this a review?

**Pull request merged**